### PR TITLE
[+] Add some initial implementations of entities.

### PR DESCRIPTION
**Pull request by ScottCTD**
- Refer to issue #4

### CODE REVIEW

File: src/main/java/billgates/entities/AbstractEntry.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+                         Attribute<String> from, Attribute<String> to, Attribute<String> location) {
+        this.id = id;
+        this.date = date;
+        this.currency = currency;
+        this.description = description;
+        this.from = from;
+        this.to = to;
+        this.location = location;
+    }
+
```
Comment (xgao28) [inline src/main/java/billgates/entities/AbstractEntry.java:27]: Consider an overload with default description of ''?

**Review by xgao28 — APPROVED** [review]

**Review by Ellen141319 — DISMISSED** [review]

**Review by xgao28 — DISMISSED** [review]

**Review by Lei-Tin — COMMENTED** [review]
Do we still need the UserFactory and stuff like that
Also, if we merge this thing up right now, how do I ensure my next pull request can be thrown on this? 
Do I cherry pick on my own branch with the database loading thing

**Comment by ScottCTD** [comment]
> Do we still need the UserFactory and stuff like that Also, if we merge this thing up right now, how do I ensure my next pull request can be thrown on this? Do I cherry pick on my own branch with the database loading thing

1. No. 2. You don't need to ensure.

**Review by Yuhan-Ut — APPROVED** [review]

**Review by CharlotteChenyy — APPROVED** [review]

File: src/main/java/billgates/entities/Entry.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package billgates.entities;
+
+import java.time.ZonedDateTime;
+
+public class Entry extends AbstractEntry {
+
+    protected Entry(Attribute<Integer> id, Attribute<ZonedDateTime> date, Attribute<Double> value,
+                 Attribute<String> currency, Attribute<String> description,
+                 Attribute<String> from, Attribute<String> to, Attribute<String> location) {
+        super(id, date, value, currency, description, from, to, location);
```
Comment (Lei-Tin) [inline src/main/java/billgates/entities/Entry.java:10]: Remember that this is a TODO

Comment (Lei-Tin) [reply src/main/java/billgates/entities/Entry.java:10]: That there is a splitter to be implemented under the entry class
Link is here: 
https://github.com/CSC207-2022F-UofT/course-project-bill-gates/pull/15/files#diff-9fddc4b7d153a3e58f867ac775ad71a92c63d3c9a6f9c08aa25d242db54e6a2fR10

**Review by xgao28 — APPROVED** [review]

**Review by Lei-Tin — APPROVED** [review]
I think the entities being implemented here suffices the purpose for the prototype! 
(Scott is the 🐐)

Just remember to: 
1. Implement the `SpliterEntry` class that is inheriting the AbstractEntry class
2. Modify `Entry.java` to incorporate the differences of an Entry with an `AbstractEntry`

**Review by xgao28 — APPROVED** [review]

**Review by Lei-Tin — APPROVED** [review]
Since there are no changes after I last reviewed, I will just approve
@ScottCTD, I have no clue why you requested my review again :P

**Pull request merged**